site/public/js/gradeable-notebook.js

```diff
@@ -79,13 +79,49 @@ function notebookAutosaveKey() {
  * Saves the current state of the notebook gradeable to localstorage.
  */
 function saveNotebookToLocal() {
-    if (typeof autosaveEnabled !== 'undefined' && autosaveEnabled) {
-        localStorage.setItem(notebookAutosaveKey(), JSON.stringify({
-            timestamp: Date.now(),
-            multiple_choice: gatherInputAnswersByType('multiple_choice'),
-            codebox: gatherInputAnswersByType('codebox'),
-        }));
-    }
+    const mc_inputs=[];
+    const short_answer_inputs = [];
+
+    //loop through multiple choice questions and save answers
+    $('.multiple_choice').each(function(){
+        let file_name='';
+        $(this).children('fieldset').each(function(){
+            file_name = $(this).data('filename');
+            if (file_name) {
+                const answers = [];
+                $(this).children('label').each(function(){
+                    //grab selected answers
+                    $(this).children('input').each(function(){
```

Comment:
```suggestion
                $(this).find('input').each(function(){
```
Small suggestion here, using find allows us to skip the looping through the labels. This is also good, because it doesn't tie the javascript code to a specific html structure.
Right now this code only works if the html structure is
```
<fieldset>
    <label>
        <input>
        <input>
        <input>
    </label>
    <label>
        <input>
        <input>
        <input>
    </label>
</fieldset>
```
but using find, it can be like this
```
<fieldset>
    ... (anything, any number of levels, could even be that input is a direct child!)
            <input>
</fieldset>
```